[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type

titaiwangms opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114886

This PR covers `ExportedProgram` to `test_fx_op_consistency.py`, which helps us identify the necessary but missing io_steps.
Next, we should refactor the tests to actually cover all ops supported by registry.

titaiwangms committed:
[ONNX] Extend

[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"




[ghstack-poisoned]

titaiwangms commented:
Blocked by #114892

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"




[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"




[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"




[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"




[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"




[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"




[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"


This PR covers `ExportedProgram` to `test_fx_op_consistency.py`, which helps us identify the necessary but missing io_steps.
Next, we should refactor the tests to actually cover all ops supported by registry.

[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"


This PR covers `ExportedProgram` to `test_fx_op_consistency.py`, which helps us identify the necessary but missing io_steps.
Next, we should refactor the tests to actually cover all ops supported by registry.

[ghstack-poisoned]

thiagocrepaldi commented on test/onnx/onnx_test_common.py:
```diff
@@ -520,6 +519,7 @@ class DecorateMeta:
         test_behavior: The behavior of the test case. [skip or xfail]
         matcher: The matcher to apply to the test case.
         enabled_if: Whether to enable test behavior. Usually used on onnx/ort version control
+        torchmodeltype: The type of the torch model. Defaults to None.
```
```suggestion
        torchmodeltype: The type of the torch model. Defaults to None.
```
At `test_fx_to_onnx_with_onnxruntime.py` this argument is named `model_type`. We should use the same name to make it uniform across the test files

thiagocrepaldi reviewed (approved):
Overall, LGTM. 2 non-blocking comments below

A minor comment is to rename `torchmodeltype` to match `test_fx_to_onnx_onnxruntime.py` naming.

A second comment is that the changes at `torch/onnx/_internal/fx/torch_export_graph_extractor.py` are (nice) fixes to `TorchExport` and could have their own PR to prevent mixing test infra improvement with exporter bug fixing.

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"


This PR covers `ExportedProgram` to `test_fx_op_consistency.py`, which helps us identify the necessary but missing io_steps.
Next, we should refactor the tests to actually cover all ops supported by registry.

[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"


This PR covers `ExportedProgram` to `test_fx_op_consistency.py`, which helps us identify the necessary but missing io_steps.
Next, we should refactor the tests to actually cover all ops supported by registry.

[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"


This PR covers `ExportedProgram` to `test_fx_op_consistency.py`, which helps us identify the necessary but missing io_steps.
Next, we should refactor the tests to actually cover all ops supported by registry.

[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"


This PR covers `ExportedProgram` to `test_fx_op_consistency.py`, which helps us identify the necessary but missing io_steps.
Next, we should refactor the tests to actually cover all ops supported by registry.

[ghstack-poisoned]

titaiwangms commented:
Blocked by unknow reason of discrepancy test result between local and CI...

titaiwangms committed:
Update on "[ONNX] Extend `test_fx_op_consistency.py` to cover ExportedProgram model type"


This PR covers `ExportedProgram` to `test_fx_op_consistency.py`, which helps us identify the necessary but missing io_steps.
Next, we should refactor the tests to actually cover all ops supported by registry.

[ghstack-poisoned]

titaiwangms commented:
@pytorchbot merge
